shared/src/prompts.rs

```diff
@@ -58,6 +58,37 @@ pub fn add_cidr(cidrs: &[Cidr], request: &AddCidrOpts) -> Result<Option<CidrCont
     )
 }
 
+/// Bring up a prompt to delete a CIDR. Returns the peer request.
+pub fn delete_cidr(cidrs: &[Cidr], request: &DeleteCidrOpts) -> Result<i64, Error> {
+    let eligible_cidrs: Vec<_> = cidrs
+        .iter()
+        .filter(|cidr| cidr.name != "innernet-server")
+        .collect();
```

Comment:
In this case, for the sake of user-friendliness, we might want to filter the `eligible_cidrs` more here to *only* show CIDRs that the client thinks can reasonably be deleted (with the server having the final say, of course).

```suggestion
/// Bring up a prompt to delete a CIDR. Returns the peer request.
pub fn delete_cidr(cidrs: &[Cidr], peers: &[Peer], request: &DeleteCidrOpts) -> Result<i64, Error> {
    let eligible_cidrs: Vec<_> = cidrs
        .iter()
        .filter(|cidr| {
            !cidrs.iter().any(|c| c.parent == cidr.id) && peers.iter().any(|p| p.cidr_id == cidr.id)
        })
        .collect();
```

^ this suggested code probably doesn't compile - just there to show roughly what I'm thinking

Reply:
I considered this as the first route, but I ran into a hiccup (although I don't remember what stopped me and caused me to just emit an error). I'll check out this code and make sure it compiles to integrate it, cause I agree not showing the ineligible cidrs at all is a better experience.